workflows/iuf/operations/vcs-upload/vcs-upload-content.yaml

```diff
@@ -0,0 +1,202 @@
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          effect: NoSchedule
+      metadata:
+        annotations:
+          sidecar.istio.io/inject: "false"
+      outputs:
+        parameters:
+          - name: secret_name
+            valueFrom:
+              path: /tmp/secret_name
+      retryStrategy:
+        limit: "2"
+        retryPolicy: "Always"
+        backoff:
+          duration: "10s" # Must be a string. Default unit is seconds. Could also be a Duration, e.g.: "2m", "6h", "1d"
+          factor: "2"
+          maxDuration: "1m"
+      script:
+        image: artifactory.algol60.net/csm-docker/stable/docker.io/portainer/kubectl-shell:latest-v1.21.1-amd64
+        command: [bash]
+        source: |
+          function sync_item() {
+            item_name="$1"
+            source_ns="$2"
+            destination_name="$3-$RANDOM"
+            destination_ns="$4"
+            if kubectl get $item_name -n $source_ns &> /dev/null; then
+              echo "Syncing $item_name from $source_ns to $destination_ns as $destination_name"
+              kubectl get $item_name -n $source_ns -o json | \
+                jq 'del(.metadata.namespace)' | \
+                jq 'del(.metadata.creationTimestamp)' | \
+                jq 'del(.metadata.resourceVersion)' | \
+                jq 'del(.metadata.selfLink)' | \
+                jq 'del(.metadata.uid)' | \
+                jq 'del(.metadata.ownerReferences)' | \
+                jq 'del(.metadata.name)' | \
+                jq '.metadata |= . + {"name":"'$destination_name'"}' | \
+                kubectl apply -n $destination_ns -f -
+                return $?
+            else
+              echo "Didn't find $item_name in the $source_ns namespace"
+              return 1
+            fi
+          }
+          sync_item secret/vcs-user-credentials services vcs-user-credentials-argo argo
+          rc=$?
+          echo $destination_name >> /tmp/secret_name
+          exit $rc
+    ## gitea-upload-content ##
+    - name: gitea-upload-content
+      nodeSelector:
+        kubernetes.io/hostname: ncn-m001
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          effect: NoSchedule
+      metadata:
+        annotations:
+          sidecar.istio.io/inject: "false"
+      inputs:
+        parameters:
+          - name: vcs_user_credentials_secret_name
+          - name: cf_import_gitea_url
+            value: "https://api-gw-service-nmn.local/vcs"
+          - name: cf_import_product_name
+            value: "{{=jsonpath(workflow.parameters.product_manifest, '$.current_product.manifest.name')}}"
+          - name: cf_import_product_version
+            value: "{{=jsonpath(workflow.parameters.product_manifest, '$.current_product.manifest.version')}}"
+          - name: cf_import_gitea_org
+            value: "{{workflow.parameters.cf_import_gitea_org}}"
+          - name: cf_import_content
+            value: /content
```

Comment:
Shouldn't this path also come from $.current_product.manifest.<blah>.vcs?

Reply:
No, this value is used at the image level. This is the docker image file structure and the python code expects all code to be in `/content` and `/results`. The VolumeMounts is where we should be reading parameter overrides or from the manifest. On the example json that was provided I did not see a vcs.path structure to point to yet.